src/docs/arch-eviction.dox

```diff
@@ -1,15 +1,108 @@
 /*! @arch_page arch-eviction Eviction
 
-Eviction represents the process or removing old data from the cache,
-writing it to disk if it is dirty. The general strategy uses a dedicated
-set of eviction threads that are tasked with identifying candidate data.
-If the data needs to be written, it is reconciled (converting the
-in-memory format to on-disk format), and then written. Clean memory
-can be freed if needed.
-
-Eviction cannot be triggered directly by API calls, but happens as a result
-of enough data being dirtied.
+@section eviction_overall Eviction
+Eviction in WiredTiger is a process to remove data that is currently not being accessed out of the
+memory to free up enough space to read data back into memory that is requested by the user but currently
+reside on the disk. If the data needs to be written, it is reconciled (converting the in-memory format
+to on-disk format), and then written.
 
 Eviction is managed using \c WT_EVICT_QUEUE structures,
 each of which contains a list of \c WT_EVICT_ENTRY structures.
+
+Eviction is composed of one eviction server, several eviction worker threads and three shared
+eviction_queues (two ordinary queues and one urgent queue).  Pages which are being marked to
+evict forcibly due to cache pressure goes on a urgent queue and the pages in the urgent queue
+take precedence over two ordinary queues.
+
+A user can pass the minimum and maximum number of eviction worker threads to be created in the
+configuration string, but however there is a mechanism to dynamically adjust right number of
+eviction worker that gave the best throughput depending on the current eviction load.
+
+- Eviction server walks the pages in the btree to find the candidates to be evicted ,sorts the
+  list of candidates and throws the 1/3 of the candidates to do an approximate LRU and pushes the
+  remaining pages into the eviction_queues.
+- Eviction worker threads goes over the eviction_queues to pop the pages and tries to evict.
+
+Eviction has to get an exclusive access to the page as there may be other threads still reading
```

Comment:
If there are other pages reading the content, the page become un-evictable, and the candidates check for that. We have to lock the page so that after having checked if the page is evictable, and starting to evict, a parallel reader doesn't start to access it.

Reply:
Fixed